### DISCUSSION TITLE
Alert when someone's typing

### DISCUSSION THREAD

**Discussion by toastyy00** [rigwild/discord-self-bot-console#51] [Q&A] [answered]
is it possible to make a alert and stop/add delay on the code from running. 
like when i use the farm xp script i want to stop the code or add some delay when someone's typing, cuz i see there's a function called "typing" on index.js.

**Comment by rigwild** [comment, marked as answer]
You can show your are typing, but you can't see others via API.

See:
- https://github.com/rigwild/discord-self-bot-console#listen-to-events-do-some-advanced-stuff
- https://discord.com/developers/docs/resources/channel#trigger-typing-indicator

**Reply by toastyy00** [reply]
what about detecting new messages from people instead of typing?

**Reply by rigwild** [reply]
You may take inspiration from https://github.com/rigwild/discord-self-bot-console/discussions/46#discussioncomment-4013171

**Reply by toastyy00** [reply]
@rigwild @DeadLyBro 
Can i make interaction with this button?
should i ask on new discussion or just in here?
![Screenshot_20221218_131544_Discord](https://user-images.githubusercontent.com/108380151/208284423-4325dc2b-09ad-4cc0-872e-e12cc5d67a9e.jpg)

**Reply by DeadLyBro** [reply]
Check before posting -> [#15](https://github.com/rigwild/discord-self-bot-console/discussions/15) 
Edit: That buttons named Components.

**Reply by toastyy00** [reply]
> Check before posting -> [#15](https://github.com/rigwild/discord-self-bot-console/discussions/15) Edit: That buttons named Components.

after looking at the post there is still no update on how components work.

**Reply by DeadLyBro** [reply]
> after looking at the post there is still no update on how components work.

We can't interaction with components, users can't send or click (but you can click manually not js).

**Reply by toastyy00** [reply]
> You can show your are typing, but you can't see others via API.
> 
> See:
> 
> * https://github.com/rigwild/discord-self-bot-console#listen-to-events-do-some-advanced-stuff
> * https://discord.com/developers/docs/resources/channel#trigger-typing-indicator

How to show the typing?

**Reply by DeadLyBro** [reply]
> > You can show your are typing, but you can't see others via API.
> > See:
> > 
> > * https://github.com/rigwild/discord-self-bot-console#listen-to-events-do-some-advanced-stuff
> > * https://discord.com/developers/docs/resources/channel#trigger-typing-indicator
> 
> How to show the typing?

Other users can see as you're typing on textbox's below. 
```js
await api.typing(channelOrThreadId);
```

And check Rigwild's this post message. (You and i quoted now.)

**Reply by toastyy00** [reply]
> > > You can show your are typing, but you can't see others via API.
> > > See:
> > > 
> > > * https://github.com/rigwild/discord-self-bot-console#listen-to-events-do-some-advanced-stuff
> > > * https://discord.com/developers/docs/resources/channel#trigger-typing-indicator
> > 
> > 
> > How to show the typing?
> 
> Other users can see as you're typing on textbox's below.
> 
> ```js
> await api.typing(channelOrThreadId);
> ```
> 
> And check Rigwild's this post message. (You and i quoted now.)

OK got it, thank you ❤️

**Reply by DeadLyBro** [reply]
I checked [that](https://discord.com/developers/docs/resources/channel#trigger-typing-indicator) and typing indicator fires a "Typing Start" gateway event and i mentioned on [this](https://github.com/rigwild/discord-self-bot-console/discussions/51#discussioncomment-4436975) message and comments...

**Reply by toastyy00** [reply]
> I checked [that](https://discord.com/developers/docs/resources/channel#trigger-typing-indicator) and typing indicator fires a "Typing Start" gateway event and i mentioned on [this](https://github.com/rigwild/discord-self-bot-console/discussions/51#discussioncomment-4436975) message and comments...

is it okay if i use only like this?
![image](https://user-images.githubusercontent.com/108380151/210969680-1318881d-c5c2-495f-b014-ebd68334d281.png)

**Reply by DeadLyBro** [reply]
> > I checked [that](https://discord.com/developers/docs/resources/channel#trigger-typing-indicator) and typing indicator fires a "Typing Start" gateway event and i mentioned on [this](https://github.com/rigwild/discord-self-bot-console/discussions/51#discussioncomment-4436975) message and comments...
> 
> is it okay if i use only like this? ![image](https://user-images.githubusercontent.com/108380151/210969680-1318881d-c5c2-495f-b014-ebd68334d281.png)

You actually sent message before typing lol.

**Reply by toastyy00** [reply]
what is your discord username?
I want to add friend req

**Reply by toastyy00** [reply]
> > > I checked [that](https://discord.com/developers/docs/resources/channel#trigger-typing-indicator) and typing indicator fires a "Typing Start" gateway event and i mentioned on [this](https://github.com/rigwild/discord-self-bot-console/discussions/51#discussioncomment-4436975) message and comments...
> > 
> > 
> > is it okay if i use only like this? ![image](https://user-images.githubusercontent.com/108380151/210969680-1318881d-c5c2-495f-b014-ebd68334d281.png)
> 
> You actually sent message before typing lol.

i'll fix it lol

**Reply by DeadLyBro** [reply]
> what is your discord username? I want to add friend req

DeadLyBro#5730

**Reply by toastyy00** [reply]
is it possible to stop when specified bot/user replied my message? @rigwild

This is my code
https://gist.github.com/toastyy00/47e9b9f862523fdcd319e2e6ccb9fcca

**Comment by DeadLyBro** [comment]
I guess you can get information with Gateway but idk how to do. ( Also you can check My Github Profile to get Discord Gateway Connection with Javascript. )

[Gateway Event > Typing Start](https://discord.com/developers/docs/topics/gateway-events#typing-start) 
![Screenshot_1](https://user-images.githubusercontent.com/31214077/208237086-94aaf0ef-7fd1-4883-9563-c0bf6612e534.png)

**Reply by toastyy00** [reply]
thank you appreciate it but I'm still newbie for that too.. ❤️

**Reply by DeadLyBro** [reply]
I think this should be work but users can't get message content with gateway. 

```js
switch (t) {
      case "TYPING_START": 
           console.log(d) // to log everything.
}
```

**Reply by DeadLyBro** [reply]
> I think this should be work but users can't get message content with gateway.
> 
> ```js
> switch (t) {
>       case "TYPING_START": 
>            console.log(d) // to log everything.
> }
> ```

I write intents wrong so now you can get message content with gateway.

**Reply by toastyy00** [reply]
> > I think this should be work but users can't get message content with gateway.
> > ```js
> > switch (t) {
> >       case "TYPING_START": 
> >            console.log(d) // to log everything.
> > }
> > ```
> 
> I write intents wrong so now you can get message content with gateway.

Heii thank you sir.. but idk how to do with it 😅

**Reply by DeadLyBro** [reply]
> Heii thank you sir.. but idk how to do with it 😅

Check my github page and look for " Discord WebSocket Pure Javascript " and copy `index.js` then add that code bottom switch (t) thing. Like this; 
```js
// other things...
switch (t) {
// my codes
}
switch (t) {
// typing start codes
}
});
```
